### DISCUSSION TITLE
<ktml:module position="[position]"> not working

### DISCUSSION THREAD

**Discussion by desbear** [joomlatools/joomlatools-pages#774] [Q&A] [answered]
I had already added a module at joomla position "nav__main". And I try to insert the <ktml:module position="nav__main"> code in the page. but the code not render to a module. Am I doing something wrong?

<img width="826" alt="截圖 2021-07-02 上午11 41 32" src="https://user-images.githubusercontent.com/545068/124217249-959aed80-db2a-11eb-994e-ff0a592236ed.png">
<img width="490" alt="截圖 2021-07-02 上午11 42 02" src="https://user-images.githubusercontent.com/545068/124217244-929ffd00-db2a-11eb-9b91-fd3e7bd4370c.png">

**Comment by manojLondhe** [comment, marked as answer]
Hi,

Make sure you have the Joomla extension installed, since 0.21 the module filter has been moved to this extension. Assuming you have following folder structure as needed by Pages,

```
joomlatools-pages
- cache/
- data/
- extensions/
- - joomla
- pages/
- templates/
- - layouts/
- - - index.html.php
- config.php
- redirects.php
```

Where extensions/joomla  is downloaded from https://github.com/joomlatools/joomlatools-pages/releases

Add code below in one of the layout file eg: templates/layouts/index.html.php

```
---
process:
   filters:
       [ext:joomla.template.filter.module]
---
```

and recheck once, this works for me.

I am using code below to load modules published at position "pages-top-menu"
`<ktml:modules position="pages-top-menu">`

**Reply by desbear** [reply]
Thank you 
I install Joomlatools-page under joomla
the folder structure is
joomla/
 -joomlatools-pages/
  --cache/
  --extensions/
  ----joomla/
  ------dispatcher/
  ------event/
  ------model/
  ------subscriber/
  ------template/
  --pages/
  ----home.html.php
  --templates/
  ----layouts/
  --------default.html.php
  --theme/
  ----css/
  ----js/
  ----images/

and also insert 
---
process:
   filters:
       [ext:joomla.template.filter.module]
---
to my default.html.php
the  home.html.php
can not show the module
<img width="421" alt="截圖 2021-07-02 下午10 43 06" src="https://user-images.githubusercontent.com/545068/124291191-e6d8ca80-db86-11eb-9da4-ac25efa13261.png">

**Reply by johanjanssens** [reply]
@desbear Will do some testing for you over the weekend and get back to you. This should be working.

**Reply by desbear** [reply]
> @desbear Will do some testing for you over the weekend and get back to you. This should be working.

I don't know why. But I just reloaded the page and the module appeared.
Thank you!!!

**Reply by johanjanssens** [reply]
Thats great news, could be a caching related issue. It's recommended when developing to do the following. In chrome, open up inspector, go to network tab and select the 'disable cache' select box.

![Screenshot 2021-07-03 at 20 22 26](https://user-images.githubusercontent.com/266640/124363590-6d031880-dc3c-11eb-91fc-6e873a476170.png)

This ensures that Pages will rebuild all it's internal caches, and will also make sure your browser cache is bypassed, it's the recommended approach when developing.
